engine/access/rpc/backend/backend_block_details.go

```diff
@@ -28,14 +30,13 @@ func (b *backendBlockDetails) GetLatestBlock(_ context.Context, isSealed bool) (
 	}
 
 	if err != nil {
-		err = rpc.ConvertStorageError(err)
-		return nil, flow.BlockStatusUnknown, err
+		// node should always have the latest block
+		return nil, flow.BlockStatusUnknown, status.Errorf(codes.Internal, "could not get latest block: %v", err)
```

Comment:
I'm reluctant to cause the node to crash as a result of an API request. This is generally something to avoid due to the DoS potential. In this case, it's unclear if any API response is valid if the blocks index in inconsistent, so I think we should at least halt the APIs.

The rpc engine currently isn't a component, so there's some additional work to do before we could throw. I'll open an issue to convert it and add some halting logic.

In practice, there are many other components on ANs that also check and throw given this condition, so the node won't silently fail.

Reply:
> In practice, there are many other components on ANs that also check and throw given this condition, so the node won't silently fail.

This is a fair point, we could add a comment along the lines to document the reasoning:
```
// In the RPC engine, if we encounter an error from the protocol state indicating state corruption,
// we should halt processing requests, but do throw an exception which might cause a crash:
//  - It is unsafe to process requests if we have an internally bad state. TODO issue
//  - We would like to avoid throwing an exception as a result of an Access API request by policy 
//    because this can cause DOS potential
//  - Since the protocol state is widely shared, we assume that in practice another component will
//    observe the protocol state error and throw an exception.
```